diff --git a/CMakeLists.txt b/CMakeLists.txt
index a8f9f25..6d5ec96 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -66,7 +66,7 @@ endif(UNIX)
 
 # Include subdirectories
 add_subdirectory(libcpuid)
-add_subdirectory(cpuid_tool)
+# add_subdirectory(cpuid_tool)
 if(LIBCPUID_BUILD_DRIVERS)
   add_subdirectory(drivers)
 endif(LIBCPUID_BUILD_DRIVERS)
diff --git a/libcpuid/CMakeLists.txt b/libcpuid/CMakeLists.txt
index 09189ef..1928036 100644
--- a/libcpuid/CMakeLists.txt
+++ b/libcpuid/CMakeLists.txt
@@ -21,7 +21,7 @@ if(UNIX)
   add_compile_definitions(_GNU_SOURCE)
 endif(UNIX)
 
-if("${MSVC_CXX_ARCHITECTURE_ID}" MATCHES "x64")
+if("${CMAKE_C_COMPILER_ARCHITECTURE_ID}" MATCHES "x64")
   list(APPEND cpuid_sources masm-x64.asm)
 endif()
 
